### DISCUSSION TITLE
How to add local folder to Dynamic Grid

### DISCUSSION THREAD

**Discussion by MartinVWWA** [SeleniumHQ/docker-selenium#2197] [Q&A]
Hello, I encountered an issue where I'm trying to add a folder to dynamically created containers, but it's not working as expected. Despite searching for solutions, I still end up with an empty folder in the Linux container or you can access a shared folder from such a container ? Has anyone else experienced a similar problem? Any assistance would be greatly appreciated. Thank you.

My docker-compose file 
`
version: "3"
services:
  node-docker:
    image: selenium/node-docker:4.18.0
    volumes:
      - /dev/shm:/dev/shm
      - ./assets:/opt/selenium/assets
      - ./config.toml:/opt/bin/config.toml        #Mount configuration file
      - /var/run/docker.sock:/var/run/docker.sock #Grid to run docker commands
      - C:\Users\Desktop\docker\SeleniumTestFiles:/home/seluser/Downloads
    depends_on:
      - selenium-hub
    environment:
      - SE_EVENT_BUS_HOST=selenium-hub
      - SE_EVENT_BUS_PUBLISH_PORT=4442
      - SE_EVENT_BUS_SUBSCRIBE_PORT=4443
      - SE_SCREEN_HEIGHT=1200
      - SE_SCREEN_WIDTH=1920
      - SE_VNC_NO_PASSWORD=1
    shm_size: 4gb      #Increase shm_size to prevent memory problems
    ports:
      - "5901:5901"   #You can change the port config if necessary`

**Comment by MarouaneBenbetka** [comment]
Your Docker Compose file includes a volume mount from a Windows path to a Linux path inside the container, which can be problematic because of path formatting and permission issues.
 
 Ensure the folder C:\Users\Desktop\docker\SeleniumTestFiles has the correct permissions for Docker to access it , and verify the paths in your config.
 
 if this answers your question mark it as answered. Good luck!

**Reply by MartinVWWA** [reply]
Thank you for your response. In this case, I have added a screenshot from the command prompt. The dynamic container, the first one, did not copy this folder. However, the second one, which starts with "hub," correctly copied my folder. However, in the case of the first one, which is built with my tests, it didn't copy anything when started. Maybe I don't understand something simple, but I don't know 
![image](https://github.com/SeleniumHQ/docker-selenium/assets/166417911/86517c00-02b0-4c4f-91e6-7962f5df8049)

**Comment by diemol** [comment]
This is not supported. It is not possible to mount directories to the child containers. What is your use case?

**Reply by MartinVWWA** [reply]
In my case, I wanted to attach a .pdf file to the process during the test. 
There is no way to add a file in your dockerfile or have access to a shared windows resource ?

**Reply by diemol** [reply]
If you need to upload a file, you can always use https://www.selenium.dev/documentation/webdriver/drivers/remote_webdriver/#uploads

**Reply by MartinVWWA** [reply]
Unfortunately, in this case I cannot, because we have a corporate selenium api which, unfortunately, is terrible at adding such things, so I will try this solution.

**Reply by diemol** [reply]
What does "corporate Selenium API" mean? Do you have your own in-house testing framework?

**Reply by MartinVWWA** [reply]
Yes, unfortunately, that's why I've been looking for other solutions, there's a strangely implanted driver which I can't touch in any way.

**Reply by diemol** [reply]
I see. Then your alternative would be to build the docker images with the file you need inside it.

**Reply by MartinVWWA** [reply]
I've just announced today that, regrettably, our project isn't compatible with dynamic Docker. Instead, I'll be creating classic Docker images, incorporating files directly during the build process. Thank you so much for your assistance :)
